lib/jsdom/living/helpers/geometry/matrixDomStringParser.js

```diff
@@ -0,0 +1,397 @@
+    this.buffer = domString;
+    this.index = 0;
+  }
+
+  is(symbol, lookahead = 0) {
+    return symbol.test(this.buffer.slice(this.index, this.index + 1 + lookahead));
+  }
+
+  get currentChar() {
+    return this.buffer[this.index];
+  }
+
+  get peek() {
+    return this.buffer[this.index + 1];
+  }
+
+  advance() {
+    this.index += 1;
+
+    if (this.index > this.buffer.length) {
+      throw new RangeError(`Advanced beyond the buffer length`);
+    }
+
+    return this.currentChar;
+  }
+
+  skipWhiteSpace() {
+    while (this.is(SYMBOL.whitespace)) {
+      this.advance();
+    }
+  }
+
+  skipComment() {
+    if (this.is(SYMBOL.commentOpen, 1)) {
+      while (this.index < this.buffer.length) {
+        this.advance();
+        if (this.is(SYMBOL.commentClose, 1)) {
+          // clear closing comments
+          this.advance();
+          this.advance();
+          break;
+        }
+      }
+    }
+  }
+
+  getFunction() {
+    let buffer = "";
+
+    while (!this.is(SYMBOL.lparen)) {
+      buffer += this.currentChar;
+      this.advance();
+    }
+
+    // discard lparen
+    this.advance();
+
+    return buffer;
+  }
+
+  getParams() {
+    let buffer = "";
+    const params = [];
+
+    while (!this.is(SYMBOL.rparen)) {
+      if (
+        !this.is(SYMBOL.whitespace) &&
+        !this.is(SYMBOL.comma)
+      ) {
+        buffer += this.currentChar;
+      } else if (buffer.length > 0) {
+        params.push(buffer);
+        buffer = "";
+      }
+
+      this.advance();
+    }
+
+    // clear remaining buffer
+    if (buffer.length > 0) {
+      params.push(buffer);
+    }
+
+    // discard rparen
+    this.advance();
+
+    return params;
+  }
+
+  parse() {
+    const functions = [];
+
+    while (this.index < this.buffer.length - 1) {
+      this.skipWhiteSpace();
+      this.skipComment();
+
+      const transformFunction = this.getFunction();
+      const transformParams = this.getParams();
+
+      functions.push([transformFunction, transformParams]);
+    }
+
+    return functions;
+  }
+}
+
+function validateParameterLength(expectedLength, params, fnName = "") {
+  const isValidLength = Array.isArray(expectedLength) ?
+    params.length >= expectedLength[0] && params.length <= expectedLength[1] :
+    params.length === expectedLength;
+
+  if (!isValidLength) {
+    throw new TypeError(`Received invalid number of parameters for function ${fnName}`);
+  }
+}
+
+// https://drafts.csswg.org/css-transforms-1/#typedef-transform-function
+// https://drafts.csswg.org/css-transforms-2/#three-d-transform-functions
+const transformFunctionNames = [
+  "matrix3d",
+  "matrix",
+  "translate3d",
+  "translate",
+  "translateX",
+  "translateY",
+  "translateZ",
+  "scale3d",
+  "scale",
+  "scaleX",
+  "scaleY",
+  "scaleZ",
+  "rotate3d",
+  "rotate",
+  "rotateX",
+  "rotateY",
+  "rotateZ",
+  "perspective",
+  "skew",
+  "skewX",
+  "skewY"
+];
+// https://www.w3.org/TR/css-values-4/#absolute-length
```

Comment:
https://drafts.csswg.org/css-values-4/#absolute-lengths

Also, let's put the CSS Values and Units stuff in a separate file, to ease reuse.